src/hooks/useCheckboxGroup.jsx

```diff
@@ -0,0 +1,176 @@
+
+	const [fields, setFields] = useState({});
+	const [errors, setErrors] = useState(() => formContext.formStateRef.current.errors[checkboxGroupName] ?? {});
+
+	/**
+	 * @function
+	 * @name registerCheckbox
+	 * @description A callback method used by a checkbox typed input to register itself to its group.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {object} checkboxRef The referenced input to register.
+	 */
+	const registerCheckbox = useCallback((checkboxRef) => {
+		if (checkboxRef.value) {
+			fieldsRef.current[checkboxGroupName][checkboxRef.value] = checkboxRef;
+		}
+	}, [fieldsRef, checkboxGroupName]);
+
+	/**
+	 * @function
+	 * @name unregisterCheckbox
+	 * @description A callback method used by a checkbox typed input to unregister itself from its parent group and form.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {string} checkboxValue The name under which the input's ref has been registered.
+	 */
+	const unregisterCheckbox = useCallback((checkboxValue) => {
+		if (fieldsRef.current[checkboxGroupName] && fieldsRef.current[checkboxGroupName][checkboxValue]) {
+			delete fieldsRef.current[checkboxGroupName][checkboxValue];
+
+			// @TODO: implement a way to remove the related field from the formState's errors object when all checkbox have unmounted.
+		}
+	}, [fieldsRef, checkboxGroupName]);
+
+	/**
+	 * @function
+	 * @name register
+	 * @description Method used to register a checkbox typed input to its parent group and form. It will return its name, options and a ref callback method.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {string}	value				The value of the input to register.
+	 * @param {bool}	[defaultChecked]	Whether the checkbox should be checked by default.
+	 *
+	 * @returns {object} The props to provide to the field instigating the call to this method.
+	 */
+	const register = useCallback(({ value, defaultChecked = false, ...additionalProps } = {}) => {
+		if (value === null || value === undefined) {
+			return undefined;
+		}
+
+		if (!fieldsRef.current[checkboxGroupName]) {
+			fieldsRef.current[checkboxGroupName] = {
+				isCheckboxGroup: true,
+				name: checkboxGroupName,
+			};
+
+			if (rules) {
+				fieldsRef.current[checkboxGroupName].rules = rules;
+			}
+		}
+
+		const inputId = `${checkboxGroupName}-${value}`;
+
+		// Determines whether this call is the first registration call made by the field or not.
+		const isInitialRegister = !fieldsRef.current[checkboxGroupName][value];
+
+		/**
+		 * Saves the reference to the {@link fieldsRef} object.
+		 * 	If it is its first registration call, we simply register its name, id, defaultChecked and options.
+		 *	If it has already been registered (eg: the form has been re-rendered), we simply update the reference
+		 *		to the input, without overriding the rest.
+		 */
+		fieldsRef.current[checkboxGroupName][value] = {
+			...(isInitialRegister
+				? { value }
+				: {
+					ref: (fieldsRef.current[checkboxGroupName][value] || {}).ref,
```

Comment:
Won't this crash if `fieldsRef.current[checkboxGroupName][value]` is falsy ?
Maybe `fieldsRef.current[checkboxGroupName][value]?.ref ?? [defaultValue]` could solve it